pkgs/top-level/aliases.nix

```diff
@@ -661,7 +661,9 @@ mapAliases ({
   nilfs_utils = nilfs-utils; # added 2018-04-25
   nix-direnv-flakes = nix-direnv;
   nix-review = nixpkgs-review; # added 2019-12-22
-  nixFlakes = nixStable; # added 2021-05-21
+  nixFlakes = nix; # added 2021-05-21
+  nixStable = nix; # added-2022-01-24
+  nixUnstable = nix-unstable; # added 2022-01-24
```

Comment:
This will cause a lot of warnings that are impossible to avoid without breaking compatibility with `nixos-22.05`. I've been guilty of this and it causes pain for out-of-tree expression maintainers as users complain about warnings.
Please move these to `all-packages.nix` with a comment suggesting to move them to `aliases` in 2023.

Did you rename these because we have a convention now? I might not have followed the news on this, but if we don't have a documented convention, we probably shouldn't rename these, as the next person could change it to their preference.

Reply:
I'm not sure I follow, why would make something an alias cause warnings? Moreover, why would the situation be better in 2023?

ISTR the convention is to use `kebab-case` for package names, but I don't remember where I got that from. I'm flexible on renaming `nixUnstable`, but IMHO `nixStable` should go; it's implied that `nix` is the stable release.

Reply:
I was under the assumption that all definitions in this file caused a warning when used. They do not. Sorry for the noise.

`nixStable` could be useful if you want to upgrade `nix` in an overlay but need to make an exception for some packages.

> `kebab-case` for package names

The docs only specify `-`s for versions (`nix-2-5`) but it seems sensible to extrapolate this to version-like qualifiers.

Reply:
This rename is totally unnecessary and we haven't even agreed upon how to name it. It will also break things where alias are disabled like my local setup or nixpkgs-review. If the rename is kept in all-packages.nix until next stable release there is some time to adjust to it. Also please move that to another PR since it is a bigger topic and we could then move on (or not) with the extraction.

Reply:
Please use underscores and do it in a separate PR. 
Sorry for the 180 on this, but the data speaks for itself, at least for version numbers. See https://github.com/NixOS/nixpkgs/pull/156857. 
Qualifiers are hard to grep for, but the version to qualifier extrapolation argument still makes sense.

Reply:
Moving the `nixUnstable` -> `nix_unstable` rename to a subsequent PR.